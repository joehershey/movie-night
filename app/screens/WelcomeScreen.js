import React from "react";
import {
  ImageBackground,
  StyleSheet,
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";

import fetch from "cross-fetch";

function WelcomeScreen(props) {
  return (
    <View style={styles.background}>
      <View style={{ margin: 30 }}>
        <Text style={[styles.message, STYLES.fontPrimary]}>
          Welcome to Movie Night!
        </Text>
      </View>

      <TouchableWithoutFeedback
        onPress={() => props.navigation.navigate("LogIn")}
      >
        <View style={[styles.loginButton, STYLES.btn]}>
          <Text style={[{ color: COLORS.primary, fontSize: 30 }]}>Log In</Text>
        </View>
      </TouchableWithoutFeedback>

      <TouchableWithoutFeedback
        onPress={() => props.navigation.navigate("SignUp")}
      >
        <View style={[styles.registerButton, STYLES.btn]}>
          <Text
            style={[
              styles.font,
              { color: COLORS.secondary, fontSize: 20, fontStyle: "italic" },
            ]}
          >
            New here? Sign up!
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  background: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: COLORS.primary,
  },
  message: {
    color: "white",
    fontSize: 40,
    textAlign: "center",
  },
  loginButton: {
    top: 10,
    width: "60%",
    height: 70,
    backgroundColor: "white",
  },
  registerButton: {
    top: 10,
    width: "60%",
    height: 70,
  },
});

export default WelcomeScreen;
