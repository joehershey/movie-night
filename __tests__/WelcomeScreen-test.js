import WelcomeScreen from "../app/screens/WelcomeScreen";
import React from 'react';
import renderer from 'react-test-renderer';

test("renders", () => {
    const tree = renderer.create(<WelcomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
})