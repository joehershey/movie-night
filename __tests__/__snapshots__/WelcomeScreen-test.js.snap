// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test entire page render renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1d1f1e",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "margin": 30,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {
            "fontFamily": "Didot",
            "fontWeight": "bold",
            "letterSpacing": 1.2,
          },
        ]
      }
    >
      Welcome to Movie Night!
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "height": 70,
          "top": 10,
          "width": "60%",
        },
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1d1f1e",
            "fontSize": 30,
          },
        ]
      }
    >
      Log In
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 70,
          "top": 10,
          "width": "60%",
        },
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          undefined,
          Object {
            "color": "#166fdb",
            "fontSize": 20,
            "fontStyle": "italic",
          },
        ]
      }
    >
      New here? Sign up!
    </Text>
  </View>
</View>
`;
